linked_list/linked_list.py

```diff
@@ -86,32 +147,71 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(N)
+    # Space Complexity: O(N)
     def reverse(self):

```

Comment:
👍  Interesting solution with the `pre` node.

However the space complexity is O(1)